qcengine/programs/mrchem.py

```diff
@@ -0,0 +1,312 @@
+                known_rsp_props = [
+                    ("dipole_moment", "vector"),
+                    ("quadrupole_moment", "tensor"),
+                    ("polarizability", "tensor"),
+                    ("magnetizability", "tensor"),
+                    ("nmr_shielding", "tensor"),
+                ]
+                computed_rsp_props = [
+                    ("properties", x, y, z)
+                    for x, z in known_rsp_props
+                    if x in mrchem_output["properties"]
+                    for y in mrchem_output["properties"][x].keys()
+                ]
+
+                # fill up extras:
+                # * under "raw_output" the whole JSON output from MRChem
+                # * under "properties" all the properties computed by MRChem
+                output_data["extras"] = {
+                    "raw_output": mrchem_json,
+                    "properties": {
+                        f"{ks[1]}": {f"{ks[2]}": _nested_get(mrchem_output, ks)} for ks in computed_rsp_props
+                    },
+                }
+
+                # fill up return_result
+                if input_model.driver == "energy":
+                    output_data["return_result"] = mrchem_output["properties"]["scf_energy"]["E_tot"]
+                elif input_model.driver == "properties":
+                    output_data["return_result"] = {
+                        f"{ks[1]}": {f"{ks[2]}": _nested_get(mrchem_output, ks)} for ks in computed_rsp_props
+                    }
+                else:
+                    raise RuntimeError(f"MRChem cannot run with {input_model.driver} driver")
+
+                compute_success = mrchem_output["success"]
+
+            else:
+                output_data["stderr"] = output["stderr"]
+                output_data["error"] = {
+                    "error_message": output["stderr"],
+                    "error_type": "execution_error",
+                }
+
+        # Dispatch errors, PSIO Errors are not recoverable for future runs
+        if compute_success is False:
+
+            if ("SIGSEV" in error_message) or ("SIGSEGV" in error_message) or ("segmentation fault" in error_message):
+                raise RandomError(error_message)
+            else:
+                raise UnknownError(error_message)
+
+        return AtomicResult(**output_data)
+
+    def build_input(self, input_model: "AtomicInput", config: "TaskConfig") -> Dict[str, Any]:
+        with popen([which("mrchem"), "--module"]) as exc:
+            exc["proc"].wait(timeout=30)
+        sys.path.append(exc["stdout"].split()[-1])
+        from mrchem import translate_input, validate
+
+        mrchemrec = {
+            "scratch_directory": config.scratch_directory,
+        }
+
+        opts = copy.deepcopy(input_model.keywords)
+
+        # Handle molecule
+        _, moldict = input_model.molecule.to_string(
+            dtype="mrchem", units=opts.get("world_unit", "bohr"), return_data=True
```

Comment:
you're probably waiting on a qcel release to be able to write a few mrchem harness tests for here. let me fix that ...